Command/AbstractSMCommand.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace winzou\Bundle\StateMachineBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+abstract class AbstractSMCommand extends ContainerAwareCommand
+{
+    /**
+     * @var array
+     */
+    protected $config;
+
+    protected function configure()
+    {
+        $this->addArgument('key', InputArgument::REQUIRED, 'A state machine key');
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        if (null === $input->getArgument('key')) {
```

Comment:
Symfony-style would be `if null !== $key then return` instead of having the rest of the method inside this if.